File: AirBnB/AirBnBClientApp/src/app/_models/Reservation.ts
import { Apartment } from './Apartment';
import { Guest } from './Guest';

export interface Reservation {
    id: number;
    status: string;
    apartment: Apartment;
    apartmentId: number;
    guest: Guest;
    guestId: number;
    date: Date;
    number: number;
    totalPrice: number;
}

File: AirBnB/AirBnB/Models/Reservation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AirBnB.Models
{
    public class Reservation
    {
        public int Id { get; set; }
        public string Status { get; set; }
        public Apartment Apartment { get; set; }
        public int ApartmentId { get; set; }
        public Guest Guest { get; set; }
        public int GuestId { get; set; }
        public DateTime Date { get; set; }
        public int Number { get; set; }
        public double TotalPrice { get; set; }

        public Reservation()
        {
            Status = ReservationStatus.Created;
        }
    }
}
